### ISSUE TITLE
program closes when opening the "find/replace in files..." dialog

### ISSUE THREAD

**Issue by Munin-** [LiMinggang/madedit-mod#285]
**Please provide the following information**

Madedit-Mod version (or branch):
0.4.16 64-bit
platform/architecture:
windows
compiler and compiler version:
Visual Studio 2017

please describe what symptom you see:
on a subsequent attempt to open the "find/replace in files..." dialog after closing it, the program crashes immediately.

what you would expect to see instead:
I'd expect madedit to open the dialog again.

and how to reproduce it:
open the dialog, set directory, file filter, do a replace, close dialog
open dialog again, if madedit didn't crash edit find/replace strings, execute, close
repeat until it crashes on choosing the menu option
for me it sometimes works up to 4 times before crashing, but often it will crash on the second attempt

**Comment by LiMinggang** [comment]
Would you please find it there is a crash dump file under the directory of MadEdit-Mod? It should named as Madedit_20180303T195345_9128.dmp, etc

**Comment by Munin-** [comment]
Oops, sorry, forgot to add it. Here is one of them (all .dmps are identical):

[00]:[000007F7653A89F9] (0)	initredirector()
[01]:[000007F7653C8686] (0)	initredirector()
[02]:[000007F7653870AA] (0)	initredirector()
[03]:[000007F765573E44] (0)	initredirector()
[04]:[000007F7655A5851] (0)	initredirector()
[05]:[000007F7655A61C2] (0)	initredirector()
[06]:[000007F7655A6524] (0)	initredirector()
[07]:[000007F7655A56ED] (0)	initredirector()
[08]:[000007F7655A5799] (0)	initredirector()
[09]:[000007F765627B18] (0)	initredirector()
[10]:[000007F7655A57BC] (0)	initredirector()
[11]:[000007F7655A6133] (0)	initredirector()
[12]:[000007F7656AA922] (0)	initredirector()
[13]:[000007F7656ABA45] (0)	initredirector()
[14]:[000007F7656B6775] (0)	initredirector()
[15]:[000007F76563ACEF] (0)	initredirector()
[16]:[000007FDB05013BE] (0)	IsThreadDesktopComposited()
[17]:[000007FDB05011D7] (0)	Ordinal2462()
[18]:[000007F765720EBC] (0)	initredirector()
[19]:[000007F7655EA105] (0)	initredirector()
[20]:[000007F7655E9F7E] (0)	initredirector()
[21]:[000007F7655EA1E4] (0)	initredirector()
[22]:[000007F765574FF7] (0)	initredirector()
[23]:[000007F7655F7E17] (0)	initredirector()
[24]:[000007F7655C2148] (0)	initredirector()
[25]:[000007F7657D6472] (0)	initredirector()
[26]:[000007FDAF3C1842] (0)	BaseThreadInitThunk()
[27]:[000007FDB068F9D1] (0)	RtlUserThreadStart()

**Comment by LiMinggang** [comment]
Doubt it's a bug of wxWidgets since I could not find any function name indicating MadEdit-Mod's. I also tried to reproduce it--I used to run Find In Files---but I could not reproduce it with Replace this time after 5 times close and open the dialog.

**Comment by LiMinggang** [comment]
Why did you close it?

**Comment by Munin-** [comment]
I didn't close it, at least not intentionally. Must have mis-clicked without noticing

**Comment by Munin-** [comment]
That you can't reproduce it indeed suggests there's something on my system madedit doesn't like - I'll see if there are .dlls missing, have a wrong version, etc.

**Comment by LiMinggang** [comment]
@Munin- not exactly. I made madedit statically linking to everything it needs so that it would not rely on anything special. And the call stack reminds me to recall the same issue I met before but not in "find/replace files". It happened without any clue to reproduce seldom that's why I suspect it's a bug of wxWidgets.

It might be gone with a new build or something unrelated but I can not sure. To fix it, I need to reproduce it constantly. However, I never found a way to do so.

**Comment by Munin-** [comment]
Yeah, no .dlls in the any of the dirs, so I can't experiment with those.

Since it happens on my machine quite often I could try to run a debug version to help find the cause, but I'd need a pre-compiled one as I can't compile anything on my machine - the compiler doesn't like Windows8 :-(

**Comment by gli-spirent** [comment]
What help do you need? I can provide you debug version of latest code.
